### PR TITLE
 Display currently open board on title bar #929 

### LINKED ISSUE #929: Display currently open board on title bar
Format: "HubTurbo V3.3.0 (Board 1)"

If there is no board open, indicate "No board open"

### PR DESCRIPTION
**Pull request by HyungJon**
Fixes #929

### CODE REVIEW

File: src/main/java/ui/UI.java [hunk 1]
```diff
@@ -585,5 +585,18 @@ public void triggerNotificationPaneAction() {
             }
         });
     }
-
+    
+    public void setTitle() {
```
Comment (dariusf) [inline src/main/java/ui/UI.java]: `updateTitle` might be a clearer name.

File: src/main/java/ui/UI.java [hunk 2]
```diff
@@ -585,5 +585,18 @@ public void triggerNotificationPaneAction() {
...
+        String openBoard = "";
+        if (prefs.getLastOpenBoard().isPresent()) {
+            openBoard = prefs.getLastOpenBoard().get();
+        } else {
+            openBoard = "No board open";
+        }
```
Comment (dariusf) [inline src/main/java/ui/UI.java]: Can be replaced with
```java
String openBoard = prefs.getLastOpenBoard().orElse("No board open");
```
Also I'm not sure about `HubTurbo V1.2.3 (No board open)`. Maybe `HubTurbo V1.2.3 (none)` (terser), or hide the parentheses entirely?

**Comment by dariusf** [comment]
Looks good. Final thing, on the title text: we can't differentiate a board named 'none' and having no board open now. :stuck_out_tongue: Is that a problem? @damithc

**Comment by HyungJon** [comment]
I've been thinking about that too, which is why I chose a name very unlikely for a board at first. Wasn't the best choice, but I guess we should use something along that line. I thought "none" is unlikely enough, but it'd be good to have a way to prevent the confusion altogether. Maybe use some special characters and add an board name input validation that forbids those characters?

**Comment by dariusf** [comment]
Yup, or hide the parentheses.

**Comment by dariusf** [comment]
@damithc do you have any opinions on this?

**Comment by damithc** [comment]
It's ok to use 'none'. It's unlikely that a user will name a board 'none'. We can also block that name during naming boards, but it can be done as a separate issue.

**Comment by damithc** [comment]
No tests?

**Comment by HyungJon** [comment]
When #951 is merged, allowing tests to access the UI instance. Then I'll be able to add ```getTitle``` so that the test can check whether the name was updated properly.

**Comment by damithc** [comment]
I see. May be merge this without tests and create a new issue for adding tests?

**Comment by HyungJon** [comment]
All right. Shall I create an issue for board name validation and test everything there?

**Comment by damithc** [comment]
Sure.

**Comment by dariusf** [comment]
Can this be in today as well?

**Comment by HyungJon** [comment]
Okay, Travis build passed. Coverage will fall since I'm creating tests in #975 .Can I have a toMerge?

**Comment by dariusf** [comment]
I'm merging this. It's passed before and won't cause any issues.

**Pull request merged**